File: backend/app/schemas/link.py
from __future__ import annotations
from pydantic import BaseModel
from typing import Optional
from app.enums import LinkStatus

class LinkCreate(BaseModel):
    supplier_id: int

class LinkOut(BaseModel):
    id: int
    supplier_id: int
    consumer_id: int
    status: LinkStatus
    supplier: Optional['SupplierOut'] = None
    consumer: Optional['UserBasic'] = None

    class Config:
        from_attributes = True

# Resolve forward references
from app.schemas.supplier import SupplierOut
from app.schemas.user import UserBasic
LinkOut.model_rebuild()

File: frontend/types/link.ts
import { LinkStatus } from '@/enums';
import { User } from './user';
import { Supplier } from './supplier';

export interface Link {
  id: number;
  consumer_id: number;
  supplier_id: number;
  status: LinkStatus;
  supplier?: Supplier;
  consumer?: User;
}

export interface LinkOut {
  id: number;
  consumer_id: number;
  supplier_id: number;
  status: LinkStatus;
  supplier?: Supplier;
  consumer?: User;
}
